pepper-apis/src/main/java/org/broadinstitute/ddp/db/TransactionWrapper.java

```diff
@@ -344,23 +366,28 @@ private static Handle openJdbiWithAuthRetry(DB db) {
         try (Handle h = openJdbiWithAuthRetry(getDB())) {
             h.useTransaction(callback);
         } catch (ConnectionException e) {
-            throw new DDPException("Unable to get a connection from data source", e);
+            throw new DDPException(COULD_NOT_GET_CONNECTION, e);
         }
     }
 
-    private HikariDataSource createDataSource(int maxConnections, String dbUrl) {
+    private HikariPool createDataSource(int maxConnections, String dbUrl, DB db) {
         HikariConfig config = new HikariConfig();
         config.setJdbcUrl(dbUrl);
         config.addDataSourceProperty("cachePrepStmts", "true");
         config.addDataSourceProperty("prepStmtCacheSize", "250");
         config.addDataSourceProperty("prepStmtCacheSqlLimit", "2048");
         config.addDataSourceProperty("cacheServerConfiguration", "true");
+        config.setInitializationFailTimeout(0); // don't fail to  create the pool if the db is not available
```

Comment:
This change was key--otherwise the pool wasn't initializing during password rotation resets.